examples/debian_snapshot/bullseye.yaml

```diff
@@ -33,3 +34,5 @@ packages:
   - "perl"
   # test cacerts() compatibility
   - "ca-certificates"
+  # test flat repository format package references
+  - "r-base-core"
```

Comment:
I think the issue with the `max depth exceeded` is not the only problem of using this package / repo for testing.

It seems that the versions move quite fast... I've tried running the tests of this PR and they initially failed because the version in the lock file didn't exist, so I had to update the lock. Ideally, the tests should be done with a snapshot repo that will "always" exist.

Also,, the R-project repo doesn't have `arm64` packages (it only has `i386` and `amd64`, and a few `all`). Thus, the tests will fail for `arm64`. I think it would be best to separate the test into another manifest file and another `BUILD` that e.g. only uses that package and only runs for `amd64`.

Or, preferably, find another flat repo that also has a snapshot... probably not an easy task! 😅 If I come up with such a repo, I'll comment here as well.